### PR TITLE
bumping some dependencies

### PR DESCRIPTION
**Pull request by brettsam**
For reference, here's the depsjson output before I updated:

```
 IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: C:\git\dotnet_6\azure-functions-host\test\WebJobs.Script.Tests\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      C:\git\dotnet_6\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

  Changed:
    - Google.Protobuf.dll: 3.15.8.0/3.15.8.0 -> 3.18.0.0/3.18.0.0
    - Grpc.AspNetCore.Server.ClientFactory.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.AspNetCore.Server.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Core.Api.dll: 2.0.0.0/2.39.1.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.Client.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.ClientFactory.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.Common.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Microsoft.CodeAnalysis.CSharp.dll: 3.3.0.0/3.300.119.46211 -> 3.11.0.0/3.1100.21.40306
    - Microsoft.CodeAnalysis.CSharp.Scripting.dll: 3.3.0.0/3.300.119.46211 -> 3.11.0.0/3.1100.21.40306
    - Microsoft.CodeAnalysis.dll: 3.3.0.0/3.300.119.46211 -> 3.11.0.0/3.1100.21.40306
    - Microsoft.CodeAnalysis.Scripting.dll: 3.3.0.0/3.300.119.46211 -> 3.11.0.0/3.1100.21.40306

  Removed:

  Added:
```

### CODE REVIEW

File: src/WebJobs.Script/GlobalSuppressions.cs [hunk 1]
```diff
@@ -8,226 +8,53 @@
 // "In Suppression File".
 // You do not need to add suppressions to this file manually.
 
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1721:PropertyNamesShouldNotMatchGetMethods", Scope = "member", Target = "Microsoft.Azure.WebJobs.Script.Binding.FunctionBinding.#Type")]
```
Comment (brettsam) [inline src/WebJobs.Script/GlobalSuppressions.cs:11]: not sure if it's new or not -- but VS lets you remove unused suppressions -- turns out a lot of these are old and no longer relevant.

File: src/WebJobs.Script/runtimeassemblies.json [hunk 2]
```diff
@@ -977,7 +977,7 @@
     },
     {
       "name": "System.IO.Abstractions",
-      "resolutionPolicy": "minorMatchOrLower"
+      "resolutionPolicy": "private"
```
Comment (brettsam) [inline src/WebJobs.Script/runtimeassemblies.json:980]: `System.IO.Abstractions` is not actually from the framework and we're several years behind on this. We shouldn't be sharing this with the function context.

File: src/WebJobs.Script/Description/DotNet/Compilation/CSharp/CSharpCompilation.cs [hunk 3]
```diff
@@ -78,7 +78,7 @@ private static string GetFullTypeName(ITypeSymbol type)
 
         private static bool IsOrUsesAssemblyType(ITypeSymbol typeSymbol, IAssemblySymbol assemblySymbol)
         {
-            if (Equals(typeSymbol.ContainingAssembly, assemblySymbol))
+            if (typeSymbol.ContainingAssembly.Equals(assemblySymbol, SymbolEqualityComparer.Default))
```
Comment (brettsam) [inline src/WebJobs.Script/Description/DotNet/Compilation/CSharp/CSharpCompilation.cs]: Bumping `CodeAnalysis` introduced [RS1024](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1024-symbols-should-be-compared-for-equality). The guidance is to use `Default` to be explicit.

**Review by mathewc — DISMISSED** [review]

**Comment by brettsam** [comment]
We weren't comfortable with those Roslyn updates -- we had at least one test failing and it required some investigation. So I backed those out -- here's the new depsjson test failure before I updated:

```
IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: C:\git\dotnet_6\azure-functions-host\test\WebJobs.Script.Tests\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      C:\git\dotnet_6\azure-functions-host\src\WebJobs.Script.WebHost\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

  Changed:
    - Google.Protobuf.dll: 3.15.8.0/3.15.8.0 -> 3.18.0.0/3.18.0.0
    - Grpc.AspNetCore.Server.ClientFactory.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.AspNetCore.Server.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Core.Api.dll: 2.0.0.0/2.39.1.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.Client.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.ClientFactory.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0
    - Grpc.Net.Common.dll: 2.0.0.0/2.39.0.0 -> 2.0.0.0/2.40.0.0

  Removed:

  Added:
```

**Review by liliankasem — APPROVED** [review]

**Comment by liliankasem** [comment]
@brettsam poke because it's been open for a month - do we still want to do this? Anything I can help with?

File: src/WebJobs.Script/runtimeassemblies.json (hunk 2 above)
Comment (brettsam) [reply src/WebJobs.Script/runtimeassemblies.json:980]: Well dang it -- I waited too long. This is now a breaking change. Gonna abandon this and we'll update the other refs in a future PR.

**Comment by brettsam** [comment]
Closing as I completely spaced on this and it now includes a breaking change that we cannot take to v4.

**Pull request closed without merging**